app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -36,19 +37,30 @@ class PatientRepository @Inject constructor(
 
   private var ongoingPatientEntry: OngoingPatientEntry = OngoingPatientEntry()
 
-  fun searchPatientsAndPhoneNumbers(query: String?): Observable<List<PatientSearchResult>> {
+  fun searchPatientsAndPhoneNumbers(query: String?, includeFuzzyNameSearch: Boolean = true): Observable<List<PatientSearchResult>> {
     if (query.isNullOrEmpty()) {
       return database.patientSearchDao()
           .recentlyUpdated100Records()
           .toObservable()
     }
 
-    return database.patientSearchDao()
-        .search(nameToSearchableForm(query!!))
-        .toObservable()
+    val actualQuery = nameToSearchableForm(query!!)
+
+    return if (actualQuery.all { it.isDigit() } || includeFuzzyNameSearch.not()) {
+      database.patientSearchDao()
+          .search(actualQuery)
+          .toObservable()
+    } else {
+      database.patientSearchDao()
+          .search(actualQuery)
+          .zipWith(database.fuzzyPatientSearchDao().searchForPatientsWithNameLike(actualQuery).toFlowable()) { searchResults, fuzzySearchResults ->
+            (fuzzySearchResults + searchResults).distinctBy { it.uuid }
+          }
+          .toObservable()
+    }
   }
 
-  fun searchPatientsAndPhoneNumbers(query: String?, assumedAge: Int): Observable<List<PatientSearchResult>> {
+  fun searchPatientsAndPhoneNumbers(query: String?, assumedAge: Int, includeFuzzyNameSearch: Boolean = true): Observable<List<PatientSearchResult>> {
```

Comment:
Should we move params into separate lines? 

Same question for the other `searchPatientsAndPhoneNumbers()` function.